src/geometry/adjoints.jl

```diff
@@ -41,13 +41,22 @@ function ChainRulesCore.rrule(::Type{Interpolations.FilledExtrapolation}, itp, f
     return y, filledextra_const_pb
 end
 
-function ChainRulesCore.rrule(::Type{SVector{N, T}}, x...) where {N, T}
-    y = SVector{N,T}(x...)
-    function svector_const_pb(Δy)
+function ChainRulesCore.rrule(::Type{SArray{D, T, ND, L}}, x...) where {D, T, ND, L}
+    y = SArray{D, T, ND, L}(x...)
```

Comment:
Triangle dispatch is generally a bad code smell, and in this case we can perfectly avoid it as we don't really need `D, T, ND, L` here.

```suggestion
function ChainRulesCore.rrule(::Type{AT}, x...) where AT<:SArray
    y = AT(x...)
```

Reply:
I had particularly kept the types because of cases like this - 
```julia
SVector(1.0, 2.0, 3.0) 
SVector((1.0, 2.0, 3.0))
```
Both of them may look the same but the good thing with my rrule is that it the first and second methods will fallback to this sort of a thing `SVector{2, Float64}((1.0, 2.0, 3.0))` always, and therefore it will consistently apply the rrule to any sort of functor/method that encounters a `<: SArray`. 
Basically what I meant to say that keeping a generic rrule will result in many places where the other pipeline will break. I am already seeing many tests failing if I remove the types.

Reply:
This is expected to be equivalent; obviously, the code is more fragile than I could expect. I don't have the bandwidth to figure it out so I'll just give it a pass.